src/moma.h

```diff
@@ -35,7 +35,15 @@ class MoMA
     double lambda_v;
     bool is_initialzied;  // only MoMA::initialze_uv() sets it to true
     bool is_solved;       // only MoMA::solve() sets it true.
-    arma::mat X;          // TODO: move to private. It is not MoMA's duty to keep a copy of X
+    arma::mat X;          // on X we perform the algorithm
+
+    arma::mat X_working;  // keep track of deflated matrices
+    arma::mat Y_working;
+
+    arma::mat X_original;  // const
```

Comment:
If these are `const`, can we put that in the type to get compiler enforcement?

Reply:
I am trying to make it const but encountering a bug. Will try again tomorrow.

Reply:
Ok - if you want to merge this PR before GSoC ends, I'm ok with that, but let's open an issue to track it, because this does seem important.

Reply:
#61